tests/safety/test_ford.py

```diff
@@ -0,0 +1,193 @@
+  STANDSTILL_THRESHOLD = 0.3
+  GAS_PRESSED_THRESHOLD = 3
+  RELAY_MALFUNCTION_BUS = 0
+  FWD_BUS_LOOKUP = {0: 2, 2: 0}
+
+  def setUp(self):
+    self.packer = None
+    raise unittest.SkipTest
+
+  # Driver brake pedal
+  def _brake_msg(self, brake):
+    # TODO: combine with pcm_status_msg since they share a CAN message
+    values = {
+      "BpedDrvAppl_D_Actl": 2 if brake else 1,
+      "CcStat_D_Actl": 5 if self.safety.get_controls_allowed() else 0,  # what to do
+    }
+    return self.packer.make_can_msg_panda("EngBrakeData", 0, values)
+
+  # Vehicle speed
+  def _speed_msg(self, speed):
+    values = {"Veh_V_ActlEng": speed}
+    return self.packer.make_can_msg_panda("EngVehicleSpThrottle2", 0, values)
+
+  # Drive throttle input
+  def _gas_msg(self, gas):
+    values = {"ApedPos_Pc_ActlArb": gas}
+    return self.packer.make_can_msg_panda("EngVehicleSpThrottle", 0, values)
+
+  # Cruise status
+  def _pcm_status_msg(self, enable):
+    # TODO: combine with _brake_msg since they share a CAN message
+    values = {
+      "CcStat_D_Actl": 5 if enable else 0,
+      "BpedDrvAppl_D_Actl": 1,  # what to do
+    }
+    return self.packer.make_can_msg_panda("EngBrakeData", 0, values)
+
+  # Steering wheel angle
+  def _angle_meas_msg(self, angle):
+    values = {"StePinComp_An_Est": angle}
+    return self.packer.make_can_msg_panda("SteeringPinion_Data", 0, values)
+
+  def _set_prev_angle(self, t):
+    self.safety.set_desired_angle_last(int(t * FORD_DEG_TO_CAN))
+
+  def _angle_meas_msg_array(self, angle):
+    for _ in range(6):
+      self._rx(self._angle_meas_msg(angle))
+
+  # Lane Centering Assist command
+  def _lkas_control_msg(self, angle, enabled):
+    values = {
+      "LatCtl_D_Rq": 1 if enabled else 0,
+      "LatCtlPath_An_Actl": clamp(angle * DEG_TO_RAD, -0.5, 0.5),  # FIXME if DBC changes
+    }
+    return self.packer.make_can_msg_panda("LateralMotionControl", 0, values)
+
+  def _acc_button_msg(self, cancel=0, resume=0, set=0):
+    values = {
+      "CcAslButtnCnclPress": cancel,
+      "CcAsllButtnResPress": resume,
+      "CcAslButtnSetPress": set,
+    }
+    return self.packer.make_can_msg_panda("Steering_Data_FD1", 0, values)
+
+
+class TestFordSteeringSafety(TestFordSafety):
+  TX_MSGS = [[MSG_STEERING_DATA_FD1, 0], [MSG_LANE_ASSIST_DATA1, 0], [MSG_LATERAL_MOTION_CONTROL, 0], [MSG_IPMA_DATA, 0]]
+  RELAY_MALFUNCTION_ADDR = MSG_LANE_ASSIST_DATA1
+  FWD_BLACKLISTED_ADDRS = {2: [MSG_LANE_ASSIST_DATA1, MSG_LATERAL_MOTION_CONTROL, MSG_IPMA_DATA]}
+
+  def setUp(self):
+    self.packer = CANPackerPanda("ford_lincoln_base_pt")
+    self.safety = libpandasafety_py.libpandasafety
+    self.safety.set_safety_hooks(Panda.SAFETY_FORD, 0)
+    self.safety.init_tests()
+
+  def test_angle_cmd_when_enabled(self):
+    # when controls are allowed, angle cmd rate limit is enforced
+    speeds = [0., 1., 5., 10., 15., 50.]
+    # angles = [-300, -100, -10, 0, 10, 100, 300]
+    angles = [-23, -10, 0, 10, 23]
```

Comment:
The DBC specifies that the range of angles we can command is only [-0.5, 0.5] radians... so this test needs to be updated for that. I also want to verify whether this is correct because 23.6 degrees isn't a huge range.